surveys/urls.py

```diff
@@ -22,88 +24,84 @@
         name='locations-list'),
 
     url(r'locations/deactivate/(?P<pk>[\w_\-]+)/$',
-        login_required(survey_views.LocationDeactivate.as_view()),
+        staff_required_custom_login(survey_views.LocationDeactivate.as_view()),
         name='locations-deactivate'),
 
     url(r'locations/(?P<pk>[\w_\-]+)/$',
-        login_required(survey_views.LocationDetail.as_view()),
+        staff_required_custom_login(survey_views.LocationDetail.as_view()),
         name='locations-detail'),
 
     url(r'study-areas/create/$',
-        login_required(survey_views.StudyAreaCreate.as_view()),
+        staff_required_custom_login(survey_views.StudyAreaCreate.as_view()),
         name='study-areas-create'),
 
     url(r'studies/create/$',
-        login_required(survey_views.StudyCreate.as_view()),
+        staff_required_custom_login(survey_views.StudyCreate.as_view()),
         name='studies-create'),
 
     url(r'studies/$',
-        login_required(survey_views.StudyList.as_view()),
+        staff_required_custom_login(survey_views.StudyList.as_view()),
         name='studies-list'),
 
     url(r'studies/deactivate/(?P<pk>[\w_\-]+)/$',
-        login_required(survey_views.StudyDeactivate.as_view()),
+        staff_required_custom_login(survey_views.StudyDeactivate.as_view()),
         name='studies-deactivate'),
 
     url(r'studies/(?P<pk>[\w_\-]+)/$',
-        login_required(survey_views.StudyDetail.as_view()),
+        staff_required_custom_login(survey_views.StudyDetail.as_view()),
         name='studies-detail'),
 
     url(r'surveys/edit/$',
-        login_required(survey_views.SurveyListEdit.as_view()),
+        staff_required_custom_login(survey_views.SurveyListEdit.as_view()),
         name='surveys-list-edit'),
 
     url(r'surveys/run/$',
         login_required(survey_views.SurveyListRun.as_view()),
         name='surveys-list-run'),
 
     url(r'surveys/create/$',
-        login_required(survey_views.SurveyCreate.as_view()),
+        staff_required_custom_login(survey_views.SurveyCreate.as_view()),
         name='surveys-create'),
 
     # Create survey element
     url(_(r'^elements/create/(?P<form_entry_id>\d+)/'
           r'(?P<form_element_plugin_uid>[\w_\-]+)/$'),
-        view=fobi.views.add_form_element_entry,
+        staff_required_custom_login(fobi.views.add_form_element_entry),
         name='fobi.add_form_element_entry'),
 
     # Edit survey element
     url(_(r'^elements/edit/(?P<form_element_entry_id>\d+)/$'),
-        view=fobi.views.edit_form_element_entry,
+        staff_required_custom_login(fobi.views.edit_form_element_entry),
         name='fobi.edit_form_element_entry'),
 
     # Delete survey element
     url(_(r'^elements/delete/(?P<form_element_entry_id>\d+)/$'),
-        view=fobi.views.delete_form_element_entry,
+        staff_required_custom_login(fobi.views.delete_form_element_entry),
         name='fobi.delete_form_element_entry'),
 
     # Create survey handler
```

Comment:
Looks like the following views (add, edit, and delete survey handlers) are not protected, do we need to protect those?

Reply:
Good catch! They're never exposed (Fobi offers the ability to let users add different data handlers. Instead, we've written a custom one and add it automatically to new surveys) so it's not a huge risk, but won't hurt!